bootstrap/main.go

```diff
@@ -0,0 +1,209 @@
+	args.Int32("timeout", 0, "Timeout in minutes if set")
+
+	// TODO(bentheelder): remove these?
+	args.String("pull", "", "Deprecated, use --repo=k8s.io/foo=master:abcd,12:ef12,45:ff65")
+	args.String("branch", "", "Deprecated, use --repo=k8s.io/foo=master")
+
+	args.StringArray("repo", []string{},
+		"Fetch the specified repositories, with the first one considered primary")
+
+	args.Bool("bare", false, "Do not check out a repository")
+	args.Lookup("bare").NoOptDefVal = "true" // allows using --bare
+
+	// NOTE: this arg is required (set above)
+	args.String("job", "", "Name of the job to run")
+
+	args.String("upload", "", "Upload results here if set, requires --service-account")
+	args.String("service-account", "", "Activate and use path/to/service-account.json if set.")
+	args.String("ssh", "", "Use the ssh key to fetch the repository instead of https if set.")
+	args.String("git-cache", "", "Location of the git cache.")
+
+	args.Bool("clean", false, "Clean the git repo before running tests.")
+	args.Lookup("clean").NoOptDefVal = "true" // allows using --clean
+
+	// parse then check required flags
+	args.Parse(arguments)
+	for _, arg := range requiredFlags {
+		if flag := args.Lookup(arg); !flag.Changed {
+			log.Printf("Flag '--%s' is required but was not set!\n", flag.Name)
+			printUsage(args)
+			os.Exit(-1)
+		}
+	}
+
+	// validate args
+	repo, _ := args.GetStringArray("repo")
+	bare, _ := args.GetBool("bare")
+	if bare == (len(repo) != 0) {
+		log.Printf("Expected --repo xor --bare, Got: --repo=%v, --bare=%v\n", repo, bare)
+		printUsage(args)
+		os.Exit(-1)
+	}
+	if len(repo) != 0 {
+		if args.Lookup("pull").Changed {
+			log.Fatal("Multi --repo does not support --pull, use --repo=R=branch,p1,p2")
+		}
+		if args.Lookup("branch").Changed {
+			log.Fatal("Multi --repo does not support --branch, use --repo=R=branch")
+		}
+	}
+	if bare {
+		if args.Lookup("pull").Changed || args.Lookup("branch").Changed {
+			log.Fatal("--pull and --branch should not be set with --bare!")
+		}
+	}
+
+	return args
+}
+
+// Helpers that get flag values or fail
+// <Insert grumbling about generics / macros here> ¯\_(ツ)_/¯
+func getFlagString(flags *flag.FlagSet, name string) string {
+	val, err := flags.GetString(name)
+	if err != nil {
+		log.Fatalf("Could not get flag '%s' %v", name, err)
+	}
+	return val
+}
+func getFlagStringArray(flags *flag.FlagSet, name string) []string {
+	val, err := flags.GetStringArray(name)
+	if err != nil {
+		log.Fatalf("Could not get flag '%s' %v", name, err)
+	}
+	return val
+}
+func getFlagBool(flags *flag.FlagSet, name string) bool {
+	val, err := flags.GetBool(name)
+	if err != nil {
+		log.Fatalf("Could not get flag '%s' %v", name, err)
+	}
+	return val
+}
+
+// Repo contains the components of git repo refs used in bootstrap
+type Repo struct {
+	Name   string
+	Branch string
+	Pull   string
+}
+
+// ParseRepos converts the refs related arguments to []Repo
```

Comment:
Please give examples of the expected formats

Reply:
Done.